sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -543,3 +543,35 @@ def test_bound_check_fixed_hyperparameter():
                         periodicity_bounds="fixed")  # seasonal component
     kernel = k1 + k2
     GaussianProcessRegressor(kernel=kernel).fit(X, y)
+
+
+@pytest.mark.parametrize('kernel', kernels)
+def test_constant_target(kernel):
+    # Test for constant target(s), the private _y_train_std attribute has
+    # a standard deviation of 1 instead of 0
+    gpr = GaussianProcessRegressor(kernel=kernel, normalize_y=True)
+
+    # create a constant target of 1
+    y = np.ones(X.shape[0])
+    expected_std = np.ones(shape=X.shape[0])
+    gpr.fit(X, y)
+
+    assert_array_equal(gpr._y_train_std, expected_std)
+
+    y_pred, y_cov = gpr.predict(X, return_cov=True)
+
+    assert_almost_equal(y_pred, y)
+    assert_almost_equal(np.diag(y_cov), 0.)
+
+    # Test multi-target data
+
+    n_samples = X.shape[0]
+    rng = np.random.RandomState(0)
+    y = np.concatenate([
+        rng.normal(size=(n_samples, 1)),  # non-constant target
+        np.full(shape=(n_samples, 1), fill_value=2)  # constant target
+    ], axis=1)
+
+    # Ensure no tracebacks
+    gpr.fit(X, y)
+    gpr.predict(X)
```

Comment:
Please call with the `return_cov=True` and check that the returned values have the expected shape and are have finite values:

```python
assert_all_finite(y_pred)
assert_all_finite(y_cov)
```

Reply:
> Please call with the return_cov=True

I get this:
```python
n_samples = X.shape[0]
    rng = np.random.RandomState(0)
    y = np.concatenate([
        rng.normal(size=(n_samples, 1)),  # non-constant target
        np.full(shape=(n_samples, 1), fill_value=2)  # constant target
    ], axis=1)

    print(X.shape, y.shape)

    # Ensure no tracebacks
    gpr.fit(X, y)
    gpr.predict(X, return_cov=True)
...
                # undo normalisation
>               y_cov = y_cov * self._y_train_std**2
E               ValueError: operands could not be broadcast together with shapes (6,6) (2,)
```

Shapes:
```python
print(X.shape, y.shape)
(6, 1) (6, 2)
```

Please let me know if you know how to fix this.

Reply:
Alright so we were indeed something wrong :)

In the main branch we have:

```python
>>> from sklearn.gaussian_process import GaussianProcessRegressor
>>> import numpy as np
>>> gpr = GaussianProcessRegressor(normalize_y=True)
>>> gpr.fit(np.random.randn(10, 5), np.random.randn(10, 2))
>>> gpr._y_train_mean
array([-0.07138434,  0.01457593])
>>> gpr._y_train_std
array([0.78389952, 0.4848555 ])
>>> gpr.predict(np.random.randn(10, 5), return_cov=True)
Traceback (most recent call last):
  File "<ipython-input-7-b81f064dd631>", line 1, in <module>
    gpr.predict(np.random.randn(10, 5), return_cov=True)
  File "/Users/ogrisel/code/scikit-learn/sklearn/gaussian_process/_gpr.py", line 355, in predict
    y_cov = y_cov * self._y_train_std**2
ValueError: operands could not be broadcast together with shapes (10,10) (2,)
```

so the bug is not specific to this PR but it means the multi-target case is not properly tested as it should... Maybe we should open a dedicated issue for `return_cov=True` on multi-target `GaussianProcessRegressor`.

Reply:
@ogrisel could this be related to #18065?

Reply:
I can comment out the part with multi-target and return_cov=True and ask to enable it in #18065 (or corresponding PR), would this be OK?

Reply:
> @ogrisel could this be related to #18065?

Yes you are right.

> I can comment out the part with multi-target and return_cov=True and ask to enable it in #18065 (or corresponding PR), would this be OK?

+1